skorecard/bucketers/bucketers.py

```diff
@@ -107,126 +105,72 @@ def __init__(
-                    for value in values:
-                        splits[value] = bucket_nr
-            else:
-                splits = binner.splits
-
-            # Deal with missing values
-            if self.missing_treatment in ["separate", "most_frequent", "most_risky", "least_risky", "neutral", "similar"]:
-                missing_bucket = None
-            if isinstance(self.missing_treatment, dict):
-                missing_bucket = self.missing_treatment.get(feature)
-
-            # Note that optbinning transform uses right=False
-            # https://github.com/guillermo-navas-palencia/optbinning/blob/396b9bed97581094167c9eb4744c2fd1fb5c7408/optbinning/binning/transformations.py#L126-L132
-            features_bucket_mapping_[feature] = BucketMapping(
-                feature_name=feature,
-                type=self.variables_type,
-                map=splits,
-                right=False,
-                specials=special,
-                missing_bucket=missing_bucket,
-            )
+    def _get_feature_splits(self, feature, X, y, X_unfiltered=None):
+        """
+        Finds the splits for a single feature.
 
-            # Calculate the bucket table
-            self.bucket_tables_[feature] = build_bucket_table(
-                X,
-                y,
-                column=feature,
-                bucket_mapping=features_bucket_mapping_.get(feature),
-            )
+        X and y have already been preprocessed, and have specials removed.
 
-            if self.missing_treatment in ["most_frequent", "most_risky", "least_risky", "neutral", "similar"]:
-                missing_bucket = self._find_missing_bucket(feature=feature)
-                # Repeat above procedure now we know the bucket distribution
-                features_bucket_mapping_[feature] = BucketMapping(
-                    feature_name=feature,
-                    type=self.variables_type,
-                    map=splits,
-                    right=False,
-                    specials=special,
-                    missing_bucket=missing_bucket,
-                )
+        Args:
+            feature (str): Name of the feature.
+            X (pd.Series): df with single column of feature to bucket
+            y (np.ndarray): array with target
+            X_unfiltered (pd.Series): df with single column of feature to bucket before any filtering was applied
 
-                # Recalculate the bucket table with the new bucket for missings
-                self.bucket_tables_[feature] = build_bucket_table(
-                    X,
-                    y,
-                    column=feature,
-                    bucket_mapping=features_bucket_mapping_.get(feature),
+        Returns:
+            splits, right (tuple): The splits (dict or array), and whether right=True or False.
+        """
+        # Normally Optbinning uses a DecisionTreeBucketer to do automatic prebinning
+        # We  require the user to pre-bucket explictly before using this.
```

Comment:
small comment, there is a double space between 'we' 'require'

Reply:
fixed in typo commit